src/main/java/Command.java

```diff
@@ -0,0 +1,187 @@
+                break;
+            case "list" :
+                ui.uiForList(tasks);
+                break;
+            case "done" : {
+                int index;
+                if (inputLine.length < 2) {
+                    throw new DukeException("", DukeExceptionType.NO_DESCRIPTION, Commands.DONE);
+                }
+                try {
+                    index = Integer.parseInt(inputLine[1]);
+                } catch (NumberFormatException e) {
+                    throw new DukeException("", DukeExceptionType.WRONG_DESCRIPTION, Commands.DONE);
+                }
+                if (index < 1 || index > tasks.getSize()) {
+                    throw new DukeException("", DukeExceptionType.WRONG_DESCRIPTION, Commands.DONE);
+                }
+                tasks.markDone(index);
+                ui.uiForDone(tasks.getTask(index));
+                storage.save(tasks);
+                break;
+            }
+            case "delete" : {
+                int index;
+                if (inputLine.length < 2) {
+                    throw new DukeException("", DukeExceptionType.NO_DESCRIPTION, Commands.DELETE);
+                }
+                try {
+                    index = Integer.parseInt(inputLine[1]);
+                } catch (NumberFormatException e) {
+                    throw new DukeException("", DukeExceptionType.WRONG_DESCRIPTION, Commands.DELETE);
+                }
+                if (index < 1 || index > tasks.getSize()) {
+                    throw new DukeException("", DukeExceptionType.WRONG_DESCRIPTION, Commands.DELETE);
+                }
+                Task task = tasks.getTask(index);
+                tasks.deleteTask(index);
+                ui.uiForDelete(tasks, task);
+                storage.save(tasks);
+                break;
+            }
+            case "todo" : {
+                if (inputLine.length < 2) {
+                    throw new DukeException("todo needs description", DukeExceptionType.NO_DESCRIPTION, Commands.TODO);
+                } else {
+                    StringBuilder description = new StringBuilder();
+                    for(int i = 1; i < inputLine.length; i++){
+                        description.append(inputLine[i]).append(" ");
+                    }
+                    Task task = new TodoTask(description.toString(), false);
+                    tasks.addTask(task);
+                    ui.uiForAdd(tasks,task);
+                    storage.save(tasks);
+                }
+
+                break;
+            }
+            case "deadline" : {
+                Task task;
+                StringBuilder description = new StringBuilder();
+                StringBuilder deadline = new StringBuilder();
+                if (inputLine.length < 2) {
+                    throw new DukeException("", DukeExceptionType.NO_DESCRIPTION, Commands.DEADLINE);
+                }
+                int index = 1;
+                while(!inputLine[index].equals("/by")){
```

Comment:
Is this method too long and hard to understand? Perhaps the logic related to parsing the user input can be abstracted out into the Parser class